featuretools/primitives/standard/transform/time_series/utils.py

```diff
@@ -236,3 +237,16 @@ def apply_rolling_agg_to_series(
             num_nans = min_periods - 1 + gap
         applied_rolled_series.iloc[range(num_nans)] = np.nan
     return applied_rolled_series.values
+
+
+def _apply_gap_for_expanding_primitives(
+    x: Series,
+    gap: Union[int, str],
+) -> Optional[Series]:
+    if isinstance(gap, int):
+        return x.shift(gap)
+    else:
+        raise NotImplementedError(
```

Comment:
I think this should be a `TypeError` instead of a `NotImplementedError`. My understanding is that `NotImplementedErrors` should typically be used with abstract base classes: https://docs.python.org/3/library/exceptions.html#NotImplementedError